.pre-commit-config.yaml

```diff
@@ -0,0 +1,47 @@
+repos:
+  - repo: https://github.com/psf/black-pre-commit-mirror
+    rev: 23.9.1
+    hooks:
+      - id: black
+        args:
+          - --quiet
+        files: ^((zwave_js_server|scripts|test)/.+)?[^/]+\.py$
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: no-commit-to-branch
+        args:
+          - --branch=master
+  - repo: local
+    hooks:
+      # Run mypy through our wrapper script in order to get the possible
+      # pyenv and/or virtualenv activated; it may not have been e.g. if
+      # committing from a GUI tool that was not launched from an activated
+      # shell.
+      - id: mypy
+        name: mypy
+        entry: scripts/run-in-env.sh mypy
+        language: script
+        types: [python]
+        require_serial: true
+        files: ^(zwave_js_server)/.+\.py$
+      - id: flake8
```

Comment:
Why do we need to run flake8 locally?

Reply:
I'm mirroring what we do in CI for testing: https://github.com/home-assistant-libs/zwave-js-server-python/blob/4c41c10328d69b30c2ac72d8e478baa073ce024b/tox.ini#L16-L23

Reply:
Right but maybe we can run flake8 via a dedicated pre-commit hook? If there are special requirements on the environment a local hook could be needed. I'm not sure what the case is for flake8.

Side note: We should move to ruff instead of flake8 longterm.